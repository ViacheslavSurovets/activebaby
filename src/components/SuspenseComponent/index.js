export { default as SuspenseComponent } from './SuspenseComponent';
