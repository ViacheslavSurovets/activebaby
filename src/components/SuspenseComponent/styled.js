import styled, { keyframes } from 'styled-components';

const spinnerAnimation = keyframes`
0% {
    transform: rotate(0deg);
}
  100% {
    transform: rotate(360deg);
}
`;


export const Spinner = styled.div`
  display: inline-block;
  position: relative;
  width: ${props => props.size}rem;
  height: ${props => props.size}rem;
    div{
      box-sizing: border-box;
      display: block;
      position: absolute;
      width: calc((${props => props.size}rem) - 20% );
      height: calc((${props => props.size}rem) - 20% );
      margin: calc((${props => props.size}rem) * 0.1 );
      border: calc((${props => props.size}rem) * 0.1 ) solid;
      border-radius: 50%;
      animation: ${ spinnerAnimation } 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
      border-color: ${ ( { theme } ) => theme.blue } transparent transparent transparent;
    }
    div:nth-child(1) {
      animation-delay: -0.45s;
    }
    div:nth-child(2) {
      animation-delay: -0.3s;
    }
    div:nth-child(3) {
      animation-delay: -0.15s;
    }
`;

export const SuspenseComponentWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
`;