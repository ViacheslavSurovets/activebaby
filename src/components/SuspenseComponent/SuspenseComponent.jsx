import React from 'react';
import { Spinner, SuspenseComponentWrapper } from './styled';


const SuspenseComponent = ( { ...rest } ) => (
  <SuspenseComponentWrapper>
    <Spinner { ...rest }>
      <div />
      <div />
      <div />
      <div />
    </Spinner>
  </SuspenseComponentWrapper>
);

export default SuspenseComponent;
