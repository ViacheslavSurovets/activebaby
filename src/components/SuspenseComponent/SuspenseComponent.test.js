import React from 'react';
import { shallow } from 'enzyme';
import SuspenseComponent from './SuspenseComponent';


describe ( 'SuspenseComponent', () => {

  it ( 'SuspenseComponent Snapshot', () => {
    expect ( shallow ( <SuspenseComponent /> ) ).toMatchSnapshot ();
  } );
} );
