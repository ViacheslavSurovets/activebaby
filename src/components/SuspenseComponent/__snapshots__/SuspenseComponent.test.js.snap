// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuspenseComponent SuspenseComponent Snapshot 1`] = `ShallowWrapper {}`;
